hystrix/src/main/java/feign/hystrix/ObservableCompletableFuture.java

```diff
@@ -0,0 +1,35 @@
+package feign.hystrix;
+
+import com.netflix.hystrix.HystrixCommand;
+import org.jvnet.animal_sniffer.IgnoreJRERequirement;
+import rx.Subscription;
+import rx.functions.Action1;
+
+import java.util.concurrent.CompletableFuture;
+
+@IgnoreJRERequirement
```

Comment:
Personally, I don't like this kind of thing.  If this code requires JDK8, then it should be in the `jdk8` module.

Reply:
Is there any strong reason for our current java 6 constraint?

Reply:
It is my understanding, from reading previous comments and documentation from @adriancole, it is the opinion of this project that core `feign` should remain JDK 6+ and that all JDK 8 features are in the `feign-java8` module.  This PR will require core `feign` to move to JDK 8, which has been called out in other PR and issues before as something we should try to avoid.

Personally, I do not have a strong opinion on this.  My questions and thoughts are an attempt to stay consistent.  I am open to having a discussion with the other contributors to see if we should revisit this decision.

Reply:
I change to jdk 8 baseline should require a major revision (ie 10.x)

Reply:
> change to jdk 8 baseline should require a major revision (ie 10.x)

LGTM =)